syntax="proto3";

message Article {
    string url=1;
    string full_text=2;
    uint64 created_at=3;
    uint64 published_at=4;
    string title=5;
    string author=6;
    string content=7;
    uint64 feed_id=8;
    string hash=9;
}
message Embedding {
    string url = 1;
    string model_name=2;
    string model_version = 3;
    repeated float embeddings=4;
}

message LatestPackage {
    repeated Article articles=1;
    repeated Embedding embeddings=2;
}

message ArticlePackage {
    repeated Article articles=1;
}

message EmbeddingPakcage {
    repeated Embedding embeddings=1;
}

message Feed {
    int64 id=1;
    string feed_url=2;
    string site_url=3;
    string title=4;
    int64 category_id=5;
    string category_title=6;
    int64 icon_id=7;
    string icon_type=8;
    bytes icon_content=9;
}

message FeedPackage {
    repeated Feed feeds=1;
}